internal/cli/serverless/authorizednetwork/update.go

```diff
@@ -0,0 +1,241 @@
+				if err != nil {
+					return err
+				}
+
+				// variables for input
+				fmt.Fprintln(h.IOStreams.Out, color.BlueString("Please input the following options"))
+
+				p := tea.NewProgram(initialUpdateInputModel())
+				inputModel, err := p.Run()
+				if err != nil {
+					return errors.Trace(err)
+				}
+				if inputModel.(ui.TextInputModel).Interrupted {
+					return util.InterruptError
+				}
+
+				displayName = inputModel.(ui.TextInputModel).Inputs[createAuthorizedNetworkField[flag.DisplayName]].Value()
+				ipRange = inputModel.(ui.TextInputModel).Inputs[createAuthorizedNetworkField[flag.IPRange]].Value()
+			} else {
+				// non-interactive mode doesn't need projectID
+				clusterID, err = cmd.Flags().GetString(flag.ClusterID)
+				if err != nil {
+					return errors.Trace(err)
+				}
+
+				displayName, err = cmd.Flags().GetString(flag.DisplayName)
+				if err != nil {
+					return errors.Trace(err)
+				}
+
+				ipRange, err = cmd.Flags().GetString(flag.IPRange)
+				if err != nil {
+					return errors.Trace(err)
+				}
+
+				targetIPRange, err = cmd.Flags().GetString(flag.TargetIPRange)
+				if err != nil {
+					return errors.Trace(err)
+				}
+			}
+
+			authorizedNetwork, err := util.ConvertToAuthorizedNetwork(ipRange, displayName)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			targetAuthorizedNetwork, err := util.ConvertToAuthorizedNetwork(targetIPRange, "")
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			existedAuthorizedNetworks, err := cloud.RetrieveAuthorizedNetworks(ctx, clusterID, d)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			for i, v := range existedAuthorizedNetworks {
+				if v.StartIpAddress == targetAuthorizedNetwork.StartIpAddress && v.EndIpAddress == targetAuthorizedNetwork.EndIpAddress {
+					existedAuthorizedNetworks = slices.Delete(existedAuthorizedNetworks, i, i+1)
+					break
+				}
+			}
+
+			authorizedNetworks := append(existedAuthorizedNetworks, authorizedNetwork)
+
+			body := &cluster.V1beta1ServerlessServicePartialUpdateClusterBody{
+				Cluster: &cluster.RequiredTheClusterToBeUpdated{
+					Endpoints: &cluster.V1beta1ClusterEndpoints{
+						Public: &cluster.EndpointsPublic{
+							AuthorizedNetworks: authorizedNetworks,
+						},
+					},
+				},
+			}
+			body.UpdateMask = AuthorizedNetworkMask
+
+			_, err = d.PartialUpdateCluster(ctx, clusterID, body)
+			if err != nil {
+				return errors.Trace(err)
+			}
+
+			_, err = fmt.Fprintln(h.IOStreams.Out, color.GreenString("authorized network %s is updated", displayName))
+			if err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+
+	updateCmd.Flags().StringP(flag.ClusterID, flag.ClusterIDShort, "", "The ID of the cluster.")
+	updateCmd.Flags().StringP(flag.IPRange, "", "", "The new IP range of the authorized network.")
+	updateCmd.Flags().StringP(flag.DisplayName, flag.DisplayNameShort, "", "The name of the authorized network.")
```

Comment:
I think we need to use `--old-start-ip-address` and `--old-end-ip-address` to match, and use -new-display-name`, `--new-start-ip-address` and `--new-end-ip-address` to replace. it's cumbersome, but I don't have better ideas. maybe `--old-display-name` could also be used to match.